src/node.rs

```diff
@@ -196,11 +196,10 @@ impl<K: KeyTrait + Clone, V: Clone> TwigNode<K, V> {
     }
 
     #[inline]
-    pub(crate) fn get_all_versions(&self) -> Vec<(V, u64, u64)> {
+    pub(crate) fn get_all_versions(&self) -> impl Iterator<Item = IterItem<'_, V>> {
         self.values
             .iter()
-            .map(|value| (value.value.clone(), value.version, value.ts))
-            .collect()
+            .map(move |value| (self.key.as_slice(), &value.value, value.version, value.ts))
```

Comment:
Currently returning the key to keep the interface uniform by returning the same IterItem everywhere. Do you think it is required here @gsserge ?

Reply:
I don't think key is required here. However, if this operation will participate in merging then we might keep the key to keep the iterators uniform - it's just a slice ref, not a copy. Otherwise let's drop the key.